### PR TITLE
Fix compliation with tracepoints

### PR DESCRIPTION
**Pull request by erthalion**
By default I'm compiling with tracepoints enabled, and found couple of missing variables. I guess they (tracepoints) were originally skipped. I would love to take a look at this closer, since I don't see any aio specific tracepoints, but for now just the first step.

### CODE REVIEW

File: src/backend/storage/buffer/bufmgr.c [hunk 1]
```diff
@@ -995,6 +996,8 @@ ReadBuffer_start(SMgrRelation smgr, char relpersistence, ForkNumber forkNum,
 	/* Make sure we will have room to remember the buffer pin */
 	ResourceOwnerEnlargeBuffers(CurrentResourceOwner);
 
+	isExtend = (blockNum == P_NEW);
```
Comment (anarazel) [inline src/backend/storage/buffer/bufmgr.c:999]: ReadBuffer_start() isn't called for file extensions, so this doesn't really make sense. Note the `Assert(blockNum != P_NEW)` above.

File: src/backend/storage/buffer/bufmgr.c [hunk 2]
```diff
@@ -1116,8 +1119,18 @@ ReadBuffer_common(SMgrRelation smgr, char relpersistence, ForkNumber forkNum,
 	BufferDesc *bufHdr;
 	Buffer		buf;
 	instr_time	io_start, io_time;
+	bool		isExtend;
+
+	isExtend = (blockNum == P_NEW);
+
+	TRACE_POSTGRESQL_BUFFER_READ_START(forkNum, blockNum,
+									   smgr->smgr_rnode.node.spcNode,
```
Comment (anarazel) [inline src/backend/storage/buffer/bufmgr.c:1127]: Can you just move that into ReadBuffer_extend() with a constant for isExtend?

**Comment by erthalion** [comment]
> I'm planning to refactor most of this a bit more, so that we don't use ReadBuffer* at all, when extending files. And I think that'll mean that the tracepoints will have to change a lot more radically, because they'll not make sense as is anymore.

Yes, that's what I expect (although I not entirely agree that they're pointless). Honestly saying this change is mostly for my convenience, to be able to experiment more with the branch and don't have annoying diffs at the same time :)

**Comment by anarazel** [comment]
There's an unrelated newline in the commit, but it doesn't really matter at this stage...

**Pull request merged**